lib/ansible/modules/storage/netapp/na_ontap_cluster.py

```diff
@@ -0,0 +1,241 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+module: na_ontap_cluster
+short_description: Create/Join ONTAP cluster. Apply license to cluster
+extends_documentation_fragment:
+    - netapp.na_ontap
+version_added: '2.6'
+author: Suhas Bangalore Shekar (bsuhas@netapp.com), Archana Ganesan (garchana@netapp.com)
+description:
+- Create or join or apply licenses to ONTAP clusters
+options:
+  state:
+    description:
+    - Whether the specified cluster should exist or not.
+    required: false
+    choices: ['present']
+    default: present
+  cluster_name:
+    description:
+    - The name of the cluster to manage.
+    required: False
+  cluster_ip_address:
+    description:
+    - IP address of cluster to be joined
+    required: False
+  license_code:
+    description:
+    - License code to be applied to the cluster
+    required: False
+  license_package:
+    description:
+    - License package name of the license to be removed
+    required: False
+  node_serial_number:
+    description:
+    - Serial number of the cluster node
+    required: False
+
+'''
+
+EXAMPLES = """
+    - name: Create cluster
+      na_ontap_cluster:
+        state: present
+        cluster_name: new_cluster
+        hostname: "{{ netapp_hostname }}"
+        username: "{{ netapp_username }}"
+        password: "{{ netapp_password }}"
+    - name: Add license from cluster
+      na_ontap_cluster:
+        state: present
+        cluster_name: FPaaS-A300-01
+        license_code: SGHLQDBBVAAAAAAAAAAAAAAAAAAA
+        hostname: "{{ netapp_hostname }}"
+        username: "{{ netapp_username }}"
+        password: "{{ netapp_password }}"
+    - name: Join cluster
+      na_ontap_cluster:
+        state: present
+        cluster_name: FPaaS-A300
+        cluster_ip_address: 10.61.184.181
+        hostname: "{{ netapp_hostname }}"
+        username: "{{ netapp_username }}"
+        password: "{{ netapp_password }}"
+"""
+
+RETURN = """
+"""
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+import ansible.module_utils.netapp as netapp_utils
+
+HAS_NETAPP_LIB = netapp_utils.has_netapp_lib()
+
+
+class NetAppONTAPCluster(object):
+    """
+    object initialize and class methods
+    """
+    def __init__(self):
+        self.argument_spec = netapp_utils.na_ontap_host_argument_spec()
+        self.argument_spec.update(dict(
+            state=dict(required=False, choices=['present'], default='present'),
+            cluster_name=dict(required=False, type='str'),
+            cluster_ip_address=dict(required=False, type='str'),
+            license_code=dict(required=False, type='str'),
```

Comment:
I believe this is only the cluster license.  If this is the case, it would be better to rename the field to cluster_license_code or something similar.